template.yaml

```diff
@@ -156,6 +200,97 @@ Resources:
                   - "states:StartExecution"
                 Resource: !GetAtt DBRestoreStateMachine.Arn
 
+  RestoreDBFromSharedSnapshotSSMDocument:
+    Type: 'AWS::SSM::Document'
+    Properties:
+      Name: Trapheus-RestoreDatabaseFromSharedSnapshot
+      DocumentFormat: YAML
+      DocumentType: Automation
+      Content:
+        schemaVersion: '0.3'
+        assumeRole: '{{ AutomationAssumeRole }}'
+        parameters:
+          SharedSnapshotARN:
+            type: String
+          NewSnapshotIdentifier:
+            type: String
+          AutomationAssumeRole:
+            type: String
+            default: ''
+          NewDatabaseIdentifier:
+            type: String
+          TargetVPCId:
+            type: String
+            description: vpc id that defines the virtual networking environment for the new restored DB instance.
+            default: ''
+        mainSteps:
+          - name: DescribeSharedSnapshot
+            action: 'aws:executeAwsApi'
+            inputs:
+              Service: rds
+              Api: DescribeDBSnapshots
+              DBSnapshotIdentifier: '{{ SharedSnapshotARN }}'
+              IncludeShared: true
+            outputs:
+              - Name: SharedSnapshotKmsKeyId
+                Selector: '$.DBSnapshots[0].KmsKeyId'
+                Type: String
+            nextStep: CreateManualCopyOfSharedSnapshot
+          - name: CreateManualCopyOfSharedSnapshot
+            action: 'aws:executeAwsApi'
+            outputs:
+              - Name: DBSnapshotIdentifier
+                Selector: $.DBSnapshot.DBSnapshotIdentifier
+                Type: String
+            inputs:
+              Service: rds
+              Api: CopyDBSnapshot
+              SourceDBSnapshotIdentifier: '{{ SharedSnapshotARN }}'
+              TargetDBSnapshotIdentifier: '{{ NewSnapshotIdentifier }}'
+              KmsKeyId: '{{ DescribeSharedSnapshot.SharedSnapshotKmsKeyId }}'
+            isEnd: false
+            onFailure: Abort
+            nextStep: WaitForAvailableStateOfNewManualSnapshot
+          - name: WaitForAvailableStateOfNewManualSnapshot
+            action: 'aws:waitForAwsResourceProperty'
+            inputs:
+              Service: rds
+              Api: DescribeDBSnapshots
+              PropertySelector: '$.DBSnapshots[0].Status'
+              DesiredValues:
+                - available
+              DBSnapshotIdentifier: '{{ CreateManualCopyOfSharedSnapshot.DBSnapshotIdentifier }}'
+            maxAttempts: 10
```

Comment:
What is the interval of time between two attempts . Does it grow exponentially ?

Reply:
i dont think its exponential - but will reverify and make changes if needed.

Reply:
looks to me it doesn't provide a way today to do an exponential backoff. What would happen here if the time  limit elapses

Reply:
if the time limit elapses - the execution will not proceed. but since its a snapshot copy - i dont think it takes much time.

Reply:
yes we can see the behaviour incrementally